src/Explorer/Controls/Settings/SettingsSubComponents/MongoIndexingPolicy/MongoIndexingPolicyComponent.tsx

```diff
@@ -0,0 +1,385 @@
+          }
+          onDiscard={() => {
+            this.props.onRevertIndexAdd(indexesToAddLength);
+          }}
+        />
+      </Stack>
+    );
+  };
+
+  private renderInitialIndexes = (): JSX.Element => {
+    const initialIndexes: MongoIndexDisplayProps[] = this.props.mongoIndexes
+      .map((mongoIndex: MongoIndex, arrayPosition: number) =>
+        this.getMongoIndexDisplayProps(mongoIndex, arrayPosition, true)
+      )
+      .filter(
+        (value: MongoIndexDisplayProps, arrayPosition: number) =>
+          !!value && !this.props.indexesToDrop.includes(arrayPosition)
+      );
+
+    return (
+      <Stack {...createAndAddMongoIndexStackProps} styles={mediumWidthStackStyles}>
+        <CollapsibleSectionComponent title="Current index(es)">
+          {
+            <>
+              <DetailsList
+                styles={customDetailsListStyles}
+                disableSelectionZone
+                items={initialIndexes}
+                columns={this.initialIndexesColumns}
+                selectionMode={SelectionMode.none}
+                onRenderRow={this.onRenderRow}
+                layoutMode={DetailsListLayoutMode.justified}
+              />
+              {this.renderIndexesToBeAdded()}
+            </>
+          }
+        </CollapsibleSectionComponent>
+      </Stack>
+    );
+  };
+
+  private renderIndexesToBeDropped = (): JSX.Element => {
+    const indexesToBeDropped: MongoIndexDisplayProps[] = this.props.indexesToDrop.map(
+      (dropIndex: number, arrayPosition: number) =>
+        this.getMongoIndexDisplayProps(this.props.mongoIndexes[dropIndex], arrayPosition, false)
+    );
+
+    return (
+      <Stack styles={mediumWidthStackStyles}>
+        <CollapsibleSectionComponent title="Index(es) to be dropped">
+          {indexesToBeDropped.length > 0 && (
+            <DetailsList
+              styles={customDetailsListStyles}
+              disableSelectionZone
+              items={indexesToBeDropped}
+              columns={this.indexesToBeDroppedColumns}
+              selectionMode={SelectionMode.none}
+              onRenderRow={this.onRenderRow}
+              layoutMode={DetailsListLayoutMode.justified}
+            />
+          )}
+        </CollapsibleSectionComponent>
+      </Stack>
+    );
+  };
+
+  private refreshIndexTransformationProgress = async () => {
+    this.setState({ isRefreshingIndexTransformationProgress: true });
+    try {
+      await this.props.refreshIndexTransformationProgress();
+    } finally {
+      this.setState({ isRefreshingIndexTransformationProgress: false });
+    }
+  };
+
+  public isIndexingTransforming = (): boolean =>
+    this.props.indexTransformationProgress !== undefined && this.props.indexTransformationProgress !== 100;
+
+  private onClickRefreshIndexingTransformationLink = async () => await this.refreshIndexTransformationProgress();
+
+  private renderIndexTransformationWarning = (): JSX.Element => {
```

Comment:
I see this pattern a few places. No need to create a reference to `let indexTransformationWarning: JSX.Element;`. Just return everywhere:

``` ts
private renderIndexTransformationWarning = (): JSX.Element => {
    if (this.state.isRefreshingIndexTransformationProgress) {
      return (
        <Stack horizontal {...mongoWarningStackProps}>
          <Text>Refreshing indexing policy update information</Text>
          <Spinner size={SpinnerSize.medium} />
        </Stack>
      );

      // index transformation progress can be 0
    } else if (this.isIndexingTransforming()) {
      const updateInfoString = "The indexing policy is being updated in the background. ";
This conversation was marked as resolved by srinathnarayanan

      if (this.props.indexTransformationProgress === 0) {
        return (
          <Text>
            {updateInfoString}
            <Link onClick={this.onClickRefreshIndexingTransformationLink}>
              {`Refresh to check if it has completed.`}
            </Link>
          </Text>
        );
      }
        return (
          <Text>
            {updateInfoString} The update is {this.props.indexTransformationProgress}% complete.
            <Link onClick={this.onClickRefreshIndexingTransformationLink}>{` Refresh to check the progress.`}</Link>
          </Text>
        );
      }
  };
```